staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -264,17 +264,26 @@ func (p *jsonPatcher) computeStrategicMergePatch(currentObject runtime.Object, _
 }
 
 func (p *jsonPatcher) firstPatchAttempt(currentObject runtime.Object, currentResourceVersion string) (runtime.Object, error) {
-	// first time through,
-	// 1. apply the patch
-	// 2. save the original and patched to detect whether there were conflicting changes on retries
+	var err error
 
-	objToUpdate := p.restPatcher.New()
+	// Encode will convert & return a versioned object in JSON.
+	// Store this JS for future use.
+	p.originalObjJS, err = runtime.Encode(p.codec, currentObject)
```

Comment:
prefer not mutating until we're sure we didn't hit an error

Reply:
This doesn't return conflict errors so an error here won't result in a retry. I will delay setting it anyway.